diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8dbdc3b..2117374 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,18 +19,22 @@ if(NOT CMAKE_CXX_STANDARD)
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
-  add_link_options("-Wl,--no-undefined")
+  if(NOT APPLE)
+    add_link_options("-Wl,--no-undefined")
+  endif()
 endif()
 
 find_package(ament_cmake REQUIRED)
 
+find_package(Threads REQUIRED)
 find_package(rcpputils REQUIRED)
 find_package(rmw REQUIRED)
 find_package(rosgraph_monitor_msgs REQUIRED)
 find_package(rosidl_runtime_cpp REQUIRED)
 find_package(rosidl_typesupport_cpp REQUIRED)
 
 add_library(${PROJECT_NAME} SHARED
   src/shim.cpp
@@ -46,6 +50,7 @@ target_link_libraries(${PROJECT_NAME} PUBLIC
   rmw::rmw
   rosidl_runtime_cpp::rosidl_runtime_cpp
   rosidl_typesupport_cpp::rosidl_typesupport_cpp
+  ${CMAKE_THREAD_LIBS_INIT}
   ${rosgraph_monitor_msgs_TARGETS}
 )
 
