diff --git a/include/grid_map_core/eigen_plugins/FunctorsPlugin.hpp b/include/grid_map_core/eigen_plugins/FunctorsPlugin.hpp
index b5c3abb..1e1c210 100644
--- a/include/grid_map_core/eigen_plugins/FunctorsPlugin.hpp
+++ b/include/grid_map_core/eigen_plugins/FunctorsPlugin.hpp
@@ -3,10 +3,14 @@
 
 #include <math.h>
 
+#ifndef EIGEN_EMPTY_STRUCT_CTOR
+#define EIGEN_EMPTY_STRUCT_CTOR(X) EIGEN_DEVICE_FUNC X() = default
+#endif
+
 template<typename Scalar>
 struct scalar_sum_of_finites_op
 {
-  EIGEN_EMPTY_STRUCT_CTOR(scalar_sum_of_finites_op)
+  EIGEN_EMPTY_STRUCT_CTOR(scalar_sum_of_finites_op);
   EIGEN_STRONG_INLINE const Scalar operator()(const Scalar & a, const Scalar & b) const
   {
     using std::isfinite;
@@ -29,7 +33,7 @@ struct functor_traits<scalar_sum_of_finites_op<Scalar>>
 template<typename Scalar>
 struct scalar_min_of_finites_op
 {
-  EIGEN_EMPTY_STRUCT_CTOR(scalar_min_of_finites_op)
+  EIGEN_EMPTY_STRUCT_CTOR(scalar_min_of_finites_op);
   EIGEN_STRONG_INLINE const Scalar operator()(const Scalar & a, const Scalar & b) const
   {
     using std::min;
@@ -53,7 +57,7 @@ struct functor_traits<scalar_min_of_finites_op<Scalar>>
 template<typename Scalar>
 struct scalar_max_of_finites_op
 {
-  EIGEN_EMPTY_STRUCT_CTOR(scalar_max_of_finites_op)
+  EIGEN_EMPTY_STRUCT_CTOR(scalar_max_of_finites_op);
   EIGEN_STRONG_INLINE const Scalar operator()(const Scalar & a, const Scalar & b) const
   {
     using std::max;
