diff --git a/CMakeLists.txt b/CMakeLists.txt
index c0ec00d..8229f65 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,14 +14,20 @@
 cmake_minimum_required(VERSION 3.8)
 project(rosgraph_monitor)
 
+if(WIN32)
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
+
 if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 17)
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic -Werror=switch)
-  add_link_options("-Wl,--no-undefined")
+  if(NOT APPLE)
+    add_link_options("-Wl,--no-undefined")
+  endif()
 endif()
 
 find_package(ament_cmake REQUIRED)
