diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,17 +15,27 @@
 cmake_minimum_required(VERSION 3.8)
 project(rmw_stats_shim)
 
+if(APPLE)
+  string(REPLACE "-Wl,--no-undefined" "" CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS}")
+  string(REPLACE "--no-undefined" "" CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS}")
+  string(REPLACE "-Wl,--no-undefined" "" CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS}")
+  string(REPLACE "--no-undefined" "" CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS}")
+endif()
+
 if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 17)
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
-  add_link_options("-Wl,--no-undefined")
+  if(NOT APPLE)
+    add_link_options("-Wl,--no-undefined")
+  endif()
 endif()
 
 find_package(ament_cmake REQUIRED)
 
+find_package(Threads REQUIRED)
 find_package(rcpputils REQUIRED)
 find_package(rmw REQUIRED)
 find_package(rosgraph_monitor_msgs REQUIRED)
 find_package(rosidl_runtime_cpp REQUIRED)
@@ -45,6 +55,7 @@
   rmw::rmw
   rosidl_runtime_cpp::rosidl_runtime_cpp
   rosidl_typesupport_cpp::rosidl_typesupport_cpp
+  Threads::Threads
   ${rosgraph_monitor_msgs_TARGETS}
 )
 
